sdks/go/pkg/beam/core/runtime/harness/harness.go

```diff
@@ -49,6 +50,7 @@ const URNMonitoringInfoShortID = "beam:protocol:monitoring_info_short_ids:v1"
 type Options struct {
 	RunnerCapabilities []string // URNs for what runners are able to understand over the FnAPI.
 	StatusEndpoint     string   // Endpoint for worker status reporting.
+	EnableDataSampling bool     // Enable data sampling feature
```

Comment:
I've spoken with Sam and noted, that the current design is because Python doesn't used Runner capabilites properly, even though they're set in the container image. That's fine.

But effectively, what I'd like then is to append the data sampling URN to the received RunnerCapabilities slice, and then use the existing map to activate the feature in harness/harness.go instead of adding a boolean field.

That way the downstream code is already keying off the correct spot long term (where the feature is enabled via the capability being present) so it can eventually be cleaned up once Dataflow is setting it properly. This allows other runners (like the local Prism runner) to eventually implement datasampling in a measured fashion.

Reply:
Done :)